### PR TITLE
handled the supervisor edgecase

### PR DESCRIPTION
**Pull request by mbogor**
#### What's this PR do?
Stops the grading script is the user is a supervisor/trainer.
#### Where should the reviewer start?
jobs/scheduling/AttendanceSync.js
#### What are the relevant tickets?
INT-194

### CODE REVIEW

**Comment by mbogor** [comment]
hold on merging; the sample data file is undergoing major changes, after those are merged in, this file needs to be refactored again.

File: jobs/scheduling/AttendanceSync.js [hunk 1]
```diff
@@ -156,6 +156,10 @@ function queryForCanvasCoursesAndAssignments(userID){
     if (!enrollmentObj.length) { 
       CONSOLE_WITH_TIME('This user has no enrollments');
       return;}
+    if(eobj[0].type=='TeacherEnrollment' || 'DesignerEnrollment'){
+      CONSOLE_WITH_TIME('THE USER ' + eobj[0].user.name + ' APPEARS TO BE A TEACHER, ABORTING GRADING');
```
Comment (jrauschenberg) [inline jobs/scheduling/AttendanceSync.js]: You use "eobj" here but "enrollmentObj" above. Which is it? Or are they really two separate variables?

File: jobs/scheduling/AttendanceSync.js [hunk 2]
```diff
@@ -156,6 +156,10 @@ function queryForCanvasCoursesAndAssignments(userID){
     if (!enrollmentObj.length) { 
       CONSOLE_WITH_TIME('This user has no enrollments');
       return;}
+    if(eobj[0].type=='TeacherEnrollment' || 'DesignerEnrollment'){
```
Comment (jrauschenberg) [inline jobs/scheduling/AttendanceSync.js]: I think you need to put an expression on both sides of the or, as in eobj[0].type=='TeacherEnrollment' || eobj[0].type=='DesignerEnrollment'

**Comment by jrauschenberg** [comment]
A couple comments. Other than that, good to merge. 👍

File: jobs/scheduling/AttendanceSync.js (hunk 1 above)
Comment (mbogor) [inline jobs/scheduling/AttendanceSync.js]: i fixed this in my 'fixed merge conflicts' commit

File: jobs/scheduling/AttendanceSync.js (hunk 2 above)
Comment (mbogor) [inline jobs/scheduling/AttendanceSync.js]: nah it works both ways, just tested it https://repl.it/C6s5

File: jobs/scheduling/AttendanceSync.js (hunk 1 above)
Comment (mbogor) [inline jobs/scheduling/AttendanceSync.js]: good looking out, fixed

File: jobs/scheduling/AttendanceSync.js (hunk 2 above)
Comment (jrauschenberg) [inline jobs/scheduling/AttendanceSync.js]: No, it doesn't seem to:

function testthis(vari){
if(vari==1 || 2){
console.log('started from the bottom now we here')
}
}

testthis(4) ==> still console.logs! check it out

Comment (mbogor) [inline jobs/scheduling/AttendanceSync.js]: u right

**Comment by tongxiang** [comment]
👍

**Pull request merged**